doc/source/groupby.rst

```diff
@@ -521,9 +521,17 @@ Transformation
 --------------
 
 The ``transform`` method returns an object that is indexed the same (same size)
-as the one being grouped. Thus, the passed transform function should return a
-result that is the same size as the group chunk. For example, suppose we wished
-to standardize the data within each group:
+as the one being grouped. The transform function must:
+
+* Return a result that is either the same size as the group chunk or
+  broadcastable to the size of the group chunk.
```

Comment:
I see you added an example below, but I would explicitly add here `(eg a scalar)` at the end of this sentence to make it clearer to the user what the typical use case is for the broadcastable result.